app/components/backup/connect.tsx

```diff
@@ -133,36 +143,65 @@ export const StorachaConnect = ({ open, onDismiss }: { open: boolean, onDismiss?
 		}
 	}
 
-  if (open && connErr) {
-    return (
-      <ConnectError
-        open={true}
-        error={connErr}
-		onDismiss={() => {
-		  setConnErr(undefined)
-		  if (onDismiss) onDismiss()
-		}}
-      />
-    )
-  }
-
-  if (open && verifying) {
-    return (
-      <Verify
-        open={true}
-        email={email}
-		onDismiss={onDismiss}
-      />
-    )
-  }
-
-  return (
-    <Connect
-      open={open}
-      email={email}
-      onEmailChange={setEmail}
-      onSubmit={handleConnectSubmit}
-	  onDismiss={onDismiss}
-    />
-  )
+	const waitForPlanSetup = async () => {
+		console.log('waiting for plan setup...')
+		try {
+			if (!client) throw new Error('missing Storacha client instance')
+			if (!account) throw new Error('missing account')
+
+			await account.plan.wait() 
+			const space = await client.createSpace(spaceName, { account })
+			await client.setCurrentSpace(space.did())
+			setSpace(space.did())
+
+			setIsStorachaAuthorized(true)
+			setIsPlanGateOpen(false)
+		} catch (err) {
+			console.error(err)
+			setConnErr(err as Error)
+		}
+	}
+
+	if (open && connErr) {
+		return (
+		<ConnectError
+			open={true}
```

Comment:
I wonder if you'll get a better animation experience if all these drawers are rendered but you control whether they are open with the boolean here i.e. you remove all the conditionals and instead make use of the `open` param e.g. `open={open && connErr}`.